src/test/javascript/spec/component/course-registration/course-registration-button/course-registration-button.component.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { ArtemisTestModule } from '../../../test.module';
+import { CourseManagementService } from 'app/course/manage/course-management.service';
+import { MockSyncStorage } from '../../../helpers/mocks/service/mock-sync-storage.service';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { of } from 'rxjs';
+import { HttpResponse } from '@angular/common/http';
+import { User } from 'app/core/user/user.model';
+import { MockProvider } from 'ng-mocks';
+import { TranslateService } from '@ngx-translate/core';
+import { AccountService } from 'app/core/auth/account.service';
+import { ProfileService } from 'app/shared/layouts/profiles/profile.service';
+import { ProfileInfo } from 'app/shared/layouts/profiles/profile-info.model';
+import { CourseRegistrationButtonComponent } from 'app/overview/course-registration/course-registration-button/course-registration-button.component';
+
+describe('CourseRegistrationButtonComponent', () => {
+    let fixture: ComponentFixture<CourseRegistrationButtonComponent>;
+    let component: CourseRegistrationButtonComponent;
+    let courseService: CourseManagementService;
+    let accountService: AccountService;
+    let profileService: ProfileService;
+    let registerForCourseStub: jest.SpyInstance;
+    let identityStub: jest.SpyInstance;
+    let getProfileInfoStub: jest.SpyInstance;
+    let onRegistrationStub: jest.SpyInstance;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [CourseRegistrationButtonComponent],
+            providers: [{ provide: LocalStorageService, useClass: MockSyncStorage }, { provide: SessionStorageService, useClass: MockSyncStorage }, MockProvider(TranslateService)],
```

Comment:
Same here. These are the providers that you should mock (the ones passed in the constructor of the Component):
```
        private accountService: AccountService,
        private courseService: CourseManagementService,
        private profileService: ProfileService,
        private alertService: AlertService,
```

Reply:
Ah, that makes so much more sense! Thanks!